src/unix/bsd/netbsdlike/openbsd/mod.rs

```diff
@@ -1530,6 +1539,12 @@ extern "C" {
     pub fn pthread_main_np() -> ::c_int;
     pub fn pthread_set_name_np(tid: ::pthread_t, name: *const ::c_char);
     pub fn pthread_stackseg_np(thread: ::pthread_t, sinfo: *mut ::stack_t) -> ::c_int;
+    pub fn pthread_spin_init(lock: *mut pthread_spinlock_t, pshared: ::c_int) -> ::c_int;
```

Comment:
if I don't mess myself, pthread_spin_ functions could be commun with at least NetBSD.